forge-gui/res/cardsfolder/v/vizier_of_the_menagerie.txt

```diff
@@ -4,5 +4,5 @@ Types:Creature Naga Cleric
 PT:3/4
 S:Mode$ Continuous | Affected$ Card.TopLibrary+YouCtrl | AffectedZone$ Library | MayLookAt$ You | Description$ You may look at the top card of your library any time.
 S:Mode$ Continuous | Affected$ Creature.TopLibrary+YouCtrl+nonLand | AffectedZone$ Library | MayPlay$ True | Description$ You may cast creature spells from the top of your library.
-S:Mode$ Continuous | Affected$ Creature.YouCtrl | AffectedZone$ Stack | AddHiddenKeyword$ May spend mana as though it were mana of any type to cast CARDNAME | Description$ You may spend mana as though it were mana of any type to cast creature spells.
+S:Mode$ ManaConvert | ValidPlayer$ You | ValidCard$ Creature.YouCtrl | ValidSa$ Spell | ManaConversion$ AnyType->AnyType | Description$ You may spend mana as though it were mana of any type to cast creature spells.
```

Comment:
there is a typo error on these scripts: engine checks for "ValidSA"

but that alone is not enough to fix the problems

Reply:
I also asked you to check how it was on old code before my changes

Reply:
"Ugh, that's indeed a problem but on master and this branch :/"

Reply:
We need to decide if this card should have different interaction outcomes together with Continuous:

- A. Play _Celestial Dawn_ first, then _Vizier_ -> Vizier would allow any color
- B. Play _Vizier_ before _Celestial Dawn_ -> the disallow timestamp is later so nonWhite colors get restricted

Currently the ordering would always lead to result of A, but I think supporting B too would make more sense?
Which would mean they should all just use ManaConvert static (with timestamp ordering support)

- CardPlayOption from MayPlay should probably also be added from its timestamp?
- And finally SA params like `ActivateIgnoreColor` get applied last? 🤔 

With that it would be cleaner to setup the current ConversionMatrix on payment chains:
- Start casting Spell A
- call `StaticAbilityManaConvert.manaConvert`
  - Start activating mana ability B
  - reset ManaPool matrix
  - call `StaticAbilityManaConvert.manaConvert` to calculate matrix for the mana ability
- after finishing paying B either restore previous matrix *OR* (which I think makes more sense) recalculate a new matrix again because statics might have changed?

Reply:
As far as I think it's that Celestial Dawn's restriction always overwrites the allow from Vizier

But I ask judges

Reply:
If I'm right and TImestamp doesn't matter, then we can just apply them in any order and the additive and restrictive would just apply 

as for the Payment Chain, i need to see if i convert all of them then it should be easier to reset them back again

Reply:
Judges so far says Can't beats Can